public/app/plugins/panel/geomap/layers/data/geojsonLayer.ts

```diff
@@ -109,9 +123,22 @@ export const geojsonLayer: MapLayerRegistryItem<GeoJSONMapperConfig> = {
       });
     }
 
+    const style = await getStyleConfigState(config.style);
+    const idToIdx = new Map<string, number>();
+
     const vectorLayer = new VectorLayer({
       source,
       style: (feature: FeatureLike) => {
+        const idx = idToIdx.get(feature.getId() as string);
+        const dims = style.dims;
+
+        if (idx != null && dims !== undefined) {
```

Comment:
Nit
```suggestion
        if (idx && dims) {
```

should check the same conditions unless we specifically care about `null` vs `undefined`